scalasql/test/src/datatypes/DataTypesTests.scala

```diff
@@ -212,5 +212,25 @@ trait DataTypesTests extends ScalaSqlSuite {
 
       }
     )
+    test("JoinNullable proper type mapping") - checker.recorded(
+      "???",
+      Text {
+        case class A[T[_]](id: T[Int], bId: T[Option[Int]])
+        object A extends Table[A]
+
+        object Custom extends Enumeration {
+          val Foo, Bar = Value
+
+          implicit def make: String => Value = withName
+          // implicit val mapper: TypeMapper[Custom.Value] = StringType.bimap(_.toString, Custom.withName)
+        }
+
+        case class B[T[_]](id: T[Int], custom: T[Custom.Value])
+        object B extends Table[B]
+        db.run(A.insert.columns(_.id := 1, _.bId := None))
+        val result = db.run(A.select.leftJoin(B)(_.id === _.id).single)
```

Comment:
Should there be an assertion (or two) about the `B` field values as well?  (`b.id` and `b.custom`)

Reply:
B suppose to be None in this case, since it's left joined, I'm not sure what do you want to assert extra here